### PR TITLE
upDate Txhash

### PR DESCRIPTION
**Pull request by zhangyi999**

### CODE REVIEW

File: src/transaction.js [hunk 1]
```diff
@@ -218,35 +218,33 @@ Transaction.prototype.toBuffer = function (buffer, initialOffset, stype) {
   if (serializeType !== Transaction.TxSerializeNoWitness) {
     writeVarInt(this.vin.length)
     this.vin.forEach(function (input) {
-      writeUInt64(input.amountin)
-      writeUInt32(input.blockheight)
-      writeUInt32(input.txindex)
-      writeVarSlice(input.script)
+      // writeUInt64(input.amountin)
```
Comment (dindinw) [inline src/transaction.js]: clean up code

File: src/transaction.js [hunk 2]
```diff
@@ -124,9 +123,9 @@ Transaction.fromBuffer = function (buffer, __noStrict) {
     vinLen = witnessLen
   }
   for (i = 0; i < vinLen; ++i) {
-    tx.vin[i].amountin = hasWitnesses ? readUInt64() : 0
-    tx.vin[i].blockheight = hasWitnesses ? readUInt32() : 0
-    tx.vin[i].txindex = hasWitnesses ? readUInt32() : 0
+    // tx.vin[i].amountin = hasWitnesses ? readUInt64() : 0
```
Comment (dindinw) [inline src/transaction.js]: clean up code

File: package.json [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
+    "qitmeer.js",
+    "qitmeer-lib",
+    "qitmeer",
+    "qitmeer-js"
   ],
   "author": "dindinw",
   "license": "ISC",
   "bugs": {
-    "url": "https://github.com/dindinw/noxjs-lib/issues"
+    "url": "https://github.com/HalalChain/qitmeer.js/issues"
```
Comment (dindinw) [inline package.json]: fix url qitmeer

File: package.json [hunk 4]
```diff
@@ -1,28 +1,30 @@
...
     "test": "npm run standard && npm run unit",
     "standard": "standard",
-    "unit": "mocha"
+    "unit": "mocha ./test/qitmeer.core.js",
+    "standfix": "standard --fix"
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/dindinw/noxjs-lib.git"
+    "url": "git+https://github.com/HalalChain/qitmeer.js"
```
Comment (dindinw) [inline package.json]: fix url

File: package.json [hunk 5]
```diff
@@ -1,28 +1,30 @@
...
+    "qitmeer-js"
   ],
   "author": "dindinw",
   "license": "ISC",
   "bugs": {
-    "url": "https://github.com/dindinw/noxjs-lib/issues"
+    "url": "https://github.com/HalalChain/qitmeer.js/issues"
   },
-  "homepage": "https://github.com/dindinw/noxjs-lib#readme",
+  "homepage": "https://github.com/HalalChain/qitmeer.js#readme",
```
Comment (dindinw) [inline package.json]: fix url

File: package.json [hunk 6]
```diff
@@ -31,10 +33,14 @@
...
     "create-hash": "^1.2.0",
     "merkle-lib": "^2.0.10",
     "randombytes": "^2.0.6",
+    "standard": "^12.0.1",
     "tiny-secp256k1": "^1.0.1",
     "varuint-bitcoin": "^1.1.0"
-  }
+  },
+  "_from": "qitmeer-js@1.0.0",
+  "_resolved": "https://registry.npm.taobao.org/qitmeer-js/download/qitmeer-js-1.0.0.tgz"
```
Comment (dindinw) [inline package.json]: why need the additional resolver ?

File: src/block.js [hunk 7]
```diff
@@ -59,8 +64,12 @@ Block.fromBuffer = function (buffer) {
   block.stateRoot = readSlice(32)
   block.difficulty = readUInt32()
   block.height = readUInt64()
-  block.timestamp = readUInt32()
-  block.nonce = readUInt64()
+  block.timestamp = readUInt64()
+  // block.nonce = readUInt64()
```
Comment (dindinw) [inline src/block.js]: clean up code

File: src/block.js [hunk 8]
```diff
@@ -59,8 +64,12 @@ Block.fromBuffer = function (buffer) {
...
+
+  // block.nonce > 2^53-1
```
Comment (dindinw) [inline src/block.js]: better comments

File: src/block.js [hunk 9]
```diff
@@ -114,20 +133,34 @@ Block.prototype.toBuffer = function (headersOnly) {
...
   writeSlice(this.parentRoot)
   writeSlice(this.txRoot)
   writeSlice(this.stateRoot)
   writeUInt32(this.difficulty)
   writeUInt64(this.height)
-  writeUInt32(this.timestamp)
-  writeUInt64(this.nonce)
-
+  writeUInt64(this.timestamp)
+  // writeUInt64(this.nonce)
```
Comment (dindinw) [inline src/block.js]: clean up code

File: src/block.js [hunk 10]
```diff
@@ -114,20 +133,34 @@ Block.prototype.toBuffer = function (headersOnly) {
...
+
+  // nonce > 2*53-1
```
Comment (dindinw) [inline src/block.js]: wrong! 2^53-1 MAX_SAFE_INTEGER

File: src/qitmeer58check.js [hunk 11]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2017-2018 The nox developers
+// Copyright 2017-2018 The Qir'me'er developers
```
Comment (dindinw) [inline src/qitmeer58check.js]: what's this?

File: src/transaction.js [hunk 12]
```diff
@@ -154,7 +153,9 @@ Transaction.prototype.byteLength = function (stype) {
     (onlyWitnesses ? 0 : this.vout.reduce(function (sum, output) { return sum + 8 + varSliceSize(output.script) }, 0)) + // amount + script
     (onlyWitnesses ? 0 : 4 + 4) + // lock-time + expire
     (hasWitnesses ? varuint.encodingLength(this.vin.length) : 0) + // the varint for witness
-    (hasWitnesses ? this.vin.reduce(function (sum, input) { return sum + 8 + 4 + 4 + varSliceSize(input.script) }, 0) : 0)
+    (hasWitnesses ? this.vin.reduce(function (sum, input) {
+      return sum + (Buffer.alloc(2).compare(input.script) === 0 ? 1 : varSliceSize(input.script))
+    }, 0) : 0)
   // amountin + blockheight + txindex + script
```
Comment (dindinw) [inline src/transaction.js]: fix the comment

File: src/transaction.js [hunk 13]
```diff
@@ -295,9 +293,9 @@ Transaction.prototype.clone = function () {
       txid: txIn.txid,
       vout: txIn.vout,
       sequence: txIn.sequence,
-      amountin: txIn.amountin || 0,
-      blockheight: txIn.blockheight || 0,
-      txindex: txIn.txindex || 0,
+      // amountin: txIn.amountin || 0,
```
Comment (dindinw) [inline src/transaction.js]: clean up code

File: package.json (hunk 6 above)
Comment (zhangyi999) [reply package.json]: cnpm i qitmeer-js

**Pull request merged**